crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/options.rs

```diff
@@ -33,6 +33,25 @@ impl UtilityClassSortingOptions {
         false
     }
 
+    pub(crate) fn match_function(&self, name: &str) -> bool {
+        let matchers = self.functions.iter().flatten();
+        let parts = name.split('.');
+        for matcher in matchers {
+            let mut matcher = matcher.split('.');
+            let mut parts = parts.clone();
+
+            let mut zip = matcher.by_ref().zip(parts.by_ref());
+            if zip.all(|(m, p)| m == "*" || m == p)
+                && matcher.next().is_none()
+                && parts.next().is_none()
+            {
+                return true;
+            }
+        }
+
+        false
+    }
```

Comment:
```suggestion
    pub(crate) fn match_function(&self, name: &str) -> bool {
        self.functions.iter().flatten().any(|matcher| {
            let mut matcher_parts = matcher.split('.');
            let mut name_parts = name.split('.').clone();

            let all_parts_match = matcher_parts
                .by_ref()
                .zip(name_parts.by_ref())
                .all(|(m, p)| m == "*" || m == p);

            all_parts_match
                && matcher_parts.next().is_none()
                && name_parts.next().is_none()
        })
    }
```
uses .any() to replace the explicit loop and manual return true. the result is functionally equivalent, with the same time and space complexity.
I prepared some testcases and  tested this on the playground, and the behavior matches the original implementation.

Reply:
Thank you for suggesting this!🥹
Committed without using redundant `.clone()`! [2e70bca](https://github.com/biomejs/biome/pull/4699/commits/2e70bca310e62695e8ff4c9bcf20890fa38afd7f)

```diff
-            let mut name_parts = name.split('.').clone();
+            let mut name_parts = name.split('.');
```